athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/common/PartitionUtil.java

```diff
@@ -0,0 +1,208 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connectors.gcs.common;
+
+import com.amazonaws.services.glue.model.Column;
+import com.amazonaws.services.glue.model.Table;
+import org.apache.arrow.vector.complex.reader.FieldReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.PARTITION_PATTERN_KEY;
+
+public class PartitionUtil
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(PartitionUtil.class);
+
+    /**
+     * Partition pattern regular expression to be used in compiling Pattern
+     */
+    private static final String PARTITION_PATTERN_REGEX = "\\{(.*?)}";
+
+    /**
+     * Pattern from a regular expression that identifies a match in a phrases to see if there is any
+     * partition key variable placeholder. A partition key variable placeholder looks something like the following:
+     * year={year}/month={month}
+     * Here, {year} and {month} are the partition key variable placeholders
+     */
+    private static final Pattern PARTITION_PATTERN = Pattern.compile(PARTITION_PATTERN_REGEX);
+
+    /**
+     * Match any alpha-num characters. Used to match VARCHAR type only partition keys
+     */
+    private static final String VARCHAR_OR_STRING_REGEX = "([a-zA-Z0-9_-\\s]+)";
+
+    private PartitionUtil()
+    {
+    }
+
+    /**
+     * Return a list of {@link AbstractMap.SimpleImmutableEntry<String, String>} instances
+     *
+     * @param table response of get table from AWS Glue
+     * @param partitionFolder      partition folder name
+     * @return List of {@link AbstractMap.SimpleImmutableEntry<String, String>} instances (column name, and value)
+     */
+    public static List<AbstractMap.SimpleImmutableEntry<String, String>> getPartitionColumnData(Table table, String partitionFolder)
+    {
+        Optional<String> optionalFolderRegex = getRegExExpression(table);
+        if (optionalFolderRegex.isPresent()) {
+            String folderRegex = optionalFolderRegex.get();
+            Pattern folderMatchPattern = Pattern.compile(folderRegex);
+            if (!folderMatchPattern.matcher(partitionFolder).matches()) {
+                return List.of();
+            }
```

Comment:
This code looks redundant, isn't it just sufficient to call `return getStoragePartitions(...)` without this check? It ends up running the same code and returning an empty collection as well, right?

Reply:
I see that you changed this code but I don't think you understood my point.
Let me be more clear, by "this code" if you look at the lines I referenced, I'm asking about the usage of the regex to match against the folder partition as a guard.
Why do we need to check if the regex matches here and then we check again doing the same thing in `getPartitionColumnData(`.
As in, if you just deleted the Pattern.compile and if statement with folderMatchPattern.matcher() and then just made a call directly into `getPartitionColumnData(`, wouldn't that just give you the same result?

Reply:
I've also provided exactly what I mean here if what I'm saying still doesn't make sense: https://github.com/awslabs/aws-athena-query-federation/pull/929/files#r1082426209

Reply:
Incorporated the changes.